src/vmm/src/builder.rs

```diff
@@ -172,14 +180,28 @@ fn create_vmm_and_vcpus(
         .map_err(VmmError::EventFd)
         .map_err(Internal)?;
 
-    let resource_allocator = ResourceAllocator::new()?;
+    let mut resource_allocator = ResourceAllocator::new()?;
 
     // Instantiate the MMIO device manager.
-    let mmio_device_manager = MMIODeviceManager::new();
+    let mut mmio_device_manager = MMIODeviceManager::new();
+
+    if boot_timer_enabled {
+        let boot_timer = crate::devices::pseudo::BootTimer::new(TimestampUs::default());
+
+        mmio_device_manager
+            .register_mmio_boot_timer(&mut resource_allocator, boot_timer)
+            .map_err(RegisterMmioDevice)?;
+    }
 
     // Instantiate ACPI device manager.
     #[cfg(target_arch = "x86_64")]
-    let acpi_device_manager = ACPIDeviceManager::new();
+    let acpi_device_manager = {
+        let cpu_container = Arc::new(Mutex::new(CpuContainer::new(
+            &mut resource_allocator,
+            vcpu_count,
+        )?));
+        ACPIDeviceManager::new(cpu_container)
+    };
```

Comment:
That'd be my fault - I suggested that we don't use an option, because unlike vmgenid, we always have a cpu container, so having it be an option looked weird to me (and using Option only to delay initialization "smells" wrong). It seemed to cause a bunch of troubles though, so I think our current code architecture forces us down the Option path :(

Reply:
Yes, I agree that is not ideal. I don't like it much either (we also always have a VMGenID). We might be able to get away with it if we ever get to restructuring the whole builder part.

Reply:
I'll revert this change then, putting it back to being optional.